src/app/InteractionModelUtilities.h

```diff
@@ -0,0 +1,115 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/util/basic-types.h>
+#include <support/BitFlags.h>
+
+namespace chip {
+namespace app {
+
+enum class CommandPathFlags : uint8_t
+{
+    kEndpointIdValid = 0x01,
+    kGroupIdValid    = 0x02,
+};
+
+struct CommandPathParams
+{
+    CommandPathParams(EndpointId aEndpointId, GroupId aGroupId, ClusterId aClusterId, CommandId aCommandId,
+                      const BitFlags<CommandPathFlags> & aFlags) :
+        mEndpointId(aEndpointId),
+        mGroupId(aGroupId), mClusterId(aClusterId), mCommandId(aCommandId), mFlags(aFlags)
+    {}
+    bool operator==(const CommandPathParams & aCommandPathParams) const
+    {
+        return ((aCommandPathParams.mClusterId == mClusterId) && (aCommandPathParams.mCommandId == mCommandId) &&
+                (mFlags.Has(aCommandPathParams.mFlags)) &&
+                (mFlags.Has(CommandPathFlags::kEndpointIdValid) ? aCommandPathParams.mEndpointId == mEndpointId : false) &&
```

Comment:
This only checks the flags in `this`. How about flags in the other one? What if mFlags.Has(kEndpointIdValid) and but `! aCommandPathParams.Has(kEndpointIdValid)` ?

Could you add unit tests for this?

Reply:
Done, thanks